java/ql/src/experimental/Security/CWE/CWE-094/SpringImplicitViewManipulation.ql

```diff
@@ -0,0 +1,67 @@
+/**
+ * @name Spring Implicit View Manipulation
+ * @description Untrusted input in a Spring View Controller can lead to RCE.
+ * @kind problem
+ * @problem.severity error
+ * @precision high
+ * @id java/spring-view-manipulation-implicit
+ * @tags security
+ *       external/cwe/cwe-094
+ */
+
+import java
+import SpringViewManipulationLib
+
+private predicate canResultInImplicitViewConversion(Method m) {
+  m.getReturnType() instanceof VoidType and
+  m.getParameter(_).(SpringRequestMappingParameter).isModelAttribute()
+  or
+  m.getReturnType() instanceof MapType
+  or
+  m.getReturnType().(RefType).hasQualifiedName("org.springframework.ui", "Model")
+  or
+  m.getReturnType() instanceof VoidType
```

Comment:
This case subsumes the first one, rendering it pointless. Are you sure this should be here?